docs/rfc/rfc-022-semi-permanent-testnet.md

```diff
@@ -0,0 +1,202 @@
+These goals could realistically be roadmapped following the launch of the MVP testnet.
+
+- The `main` testnet has more than 20 nodes
+- In addition to the `main` testnet,
+  there is at least one testnet associated with one release branch
+- The testnet no longer is simply running the Tendermint KV store;
+  now it is built on a more complex, custom application
+  that deliberately exercises a greater portion of the Tendermint stack
+- Each testnet is spread across at least two cloud providers,
+  in order to communicate over a network more closely resembling use of Tendermint in "real" chains
+- The node updates have some "jitter",
+  with some nodes updating immediately when a new build is available,
+  and others delaying up to perhaps 30-60 minutes
+- The team has published some form of dashboards that have served well for debugging,
+  which external parties can copy/modify to their needs
+- The team has produced a reference model of a log aggregation stack that external parties can use
+- Chaos engineering has begun being integrated into the testnets
+  (this could be periodic CPU limiting, deliberate network interference,
+  deliberate filesystem corruption, etc.)
+- Each testnet has at least one node running a build with the Go race detector enabled
+- The testnet contains some kind of generalized notification system built in:
+    - Tendermint code grows "watchdog" systems built in to validate things like
+      subsystems have not deadlocked; e.g. if the watchdog can't acquire and immediately release
+      a particular mutex once in every 5-minute period, it is near certain that the target
+      subsystem has deadlocked, and an alert must be sent to the engineering team.
+      (Outside of the testnet, the watchdogs could be disabled, or they could panic on failure.)
+    - The notification system does some deduplication to minimize spam on system failure
+
+### Testnet long-term vision
+
+The long-term vision includes goals that are not necessary for short- or medium-term success,
+but which would support building an increasingly stable and performant product.
+These goals would generally be beyond the one-year plan,
+and therefore they would not be part of initial planning.
+
+- There is a centralized dashboard to get a quick overview of all testnets,
+  or at least one centralized dashboard per testnet,
+  showing TBD basic information
+- Testnets include cloud spot instances which periodically and abruptly join and leave the network
+- The testnets are a heterogeneous mixture of straight VMs and Docker containers,
+  thereby more closely representing production blockchains
+- Testnets have some manner of continuous profiling,
+  so that we can produce an apples-to-apples comparison of CPU/memory cost of particular operations
+
+### Testnet non-goals
+
+There are some things we are explicitly not trying to achieve with long-lived testnets:
+
+- The testnets will NOT be maintained with any kind of on-call schedule;
```

Comment:
These sentence was not clear for me in a first read, before the following detailing. I would made it more clear, for instance, saying that there are no guarantees that testnets will be always online/available/operative.

Reply:
I think what @mark-rushakoff was trying to talk about here was the team's level of responsibility in looking after the testnet. We don't expect people to work after hours or over weekends to ensure that the testnet's available.

But a side effect of that definitely is that the testnets may not always be available.

Reply:
Hopefully clarified in 3e1d75805.